docs/src/test-sharding-js.md

```diff
@@ -0,0 +1,169 @@
+---
+id: test-sharding
+title: "Running tests on multiple machines"
+---
+
+Playwright Test runs tests in [parallel](/test-parallel.md) and strives for optimal utilization of all accessible CPU cores on your machine. As the number of tests in your suite grows, you can improve execution speed by running them on several machines simultaneously.
+
+## Sharding tests between multiple machines
+
+Playwright Test can shard a test suite, so that it can be executed on multiple machines. For that, pass --shard=x/y to the command line. For example, to split the suite into three shards, each running one third of the tests:
+
+npx playwright test --shard=1/3
+npx playwright test --shard=2/3
+npx playwright test --shard=3/3
+
+That way your test suite completes 3 times faster.
+
+### GitHub Actions sharding example
+
+One of the easiest ways to shard Playwright tests across multiple machines is by using GitHub Actions matrix strategy. For example, you can configure a job to run your tests on 4 machines in parallel like this:
+
+```yaml
+jobs:
+  test:
+    strategy:
+      matrix:
+        shard: [1/4, 2/4, 3/4, 4/4]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - name: Install dependencies
+      run: npm ci
+    - name: Install Playwright browsers
+      run: npx playwright install
+
+    - name: Run Playwright tests
+      run: npx playwright test --shard ${{ matrix.shard }}
+```
+
+## Creating combined report in GitHub Actions
+
+Each test shard will have its own test report. If you want to have a combined report showing all tests results from all shards you need to add a job that will merge individual shard reports.
+
+### Configuring shard reporting
+
+We start by adding `blob` reporter to the config:
+
+```js
+export default defineConfig({
+  testDir: './tests',
+  reporter: [['blob', { outputDir: 'blob-report' }]],
+});
+```
+
+Blob report contains information about all the tests that were run and their results as well as all test attachments such as traces and screenshot diffs. Blob reports can be merged and converted to any other Playwright report.
+
+### Uploading shard report
+
+To merge individual reports they need to be copied into a shared location. GitHub Actions Artifacts is a convenient mechanism that lets you do that. By adding the following step after the test execution we upload blob report from each shard into GitHub Actions Artifact with name`blob-report-${{ github.run_attempt }}` (this is essentially a shared directory where each shard will copy its report to):
```

Comment:
I'd move `By adding ....` sentence after the snippet, as an optional explanation.

We should also mention somehow, that we do not upload html anymore. Maybe "replace this upload with that upload"?